src/plugins/telemetry/public/services/telemetry_service.test.ts

```diff
@@ -232,6 +232,19 @@ describe('TelemetryService', () => {
       expect(telemetryService.getUserShouldSeeOptInNotice()).toBe(false);
     });
 
+    it('should return true when optIn: null even when previously seen', () => {
+      const telemetryService = mockTelemetryService({
+        config: {
+          userCanChangeSettings: true,
+          telemetryNotifyUserAboutOptInDefault: false,
+          optIn: null,
+        },
+      });
+      expect(telemetryService.config.userCanChangeSettings).toBe(true);
```

Comment:
unrelated to this PR and non-blocker: is there a reason `config` is exposed on the `telemetryService`? Seems that the `useCanChangeSettings` is the same value just synced from `config`?

Reply:
yeah... there is/was a bit of a nightmare config vs. _satinized_ config (especially for the `optIn` value, where we differentiate the actual value and the expected behaviour, and both are used in different places)... 

We def need to revisit this. I bet we can remove many of the public APIs in this service 😇.